app/grandchallenge/components/tasks.py

```diff
@@ -822,17 +844,28 @@ def handle_event(*, event, backend):  # noqa: C901
         )
 
 
-@acks_late_2xlarge_task
+@acks_late_2xlarge_task(retry_on=(LockNotAcquiredException,))
+@transaction.atomic
 def parse_job_outputs(
     *, job_pk: uuid.UUID, job_app_label: str, job_model_name: str, backend: str
 ):
-    job = get_model_instance(
-        pk=job_pk, app_label=job_app_label, model_name=job_model_name
+    model = apps.get_model(app_label=job_app_label, model_name=job_model_name)
+    job_queryset = model.objects.filter(pk=job_pk).select_for_update(
+        nowait=True
     )
+
+    try:
+        job = job_queryset.get()
+    except OperationalError as error:
+        raise LockNotAcquiredException from error
+
     executor = job.get_executor(backend=backend)
 
     if job.status == job.EXECUTED and not job.outputs.exists():
         job.update_status(status=job.PARSING)
+    elif job.status in (job.SUCCESS, job.FAILURE, job.CANCELLED):
```

Comment:
Why was this check added, does the same thing as the existing else as far as I can tell. Again, don't think these tasks needed to be addressed.